app/scripts/lib/setupSentry.js

```diff
@@ -146,31 +154,125 @@ export default function setupSentry({ release, getState }) {
+function hideUrlIfNotInternal(url) {
+  const re = /^(chrome-extension|moz-extension):\/\//u;
+  if (!url.match(re)) {
+    return '';
+  }
+  return url;
+}
+
+/**
+ * Receives a Sentry breadcrumb object and potentially removes urls
+ * from its `data` property, it particular those possibly found at
+ * data.from, data.to and data.url
+ *
+ * @param {object} breadcrumb - A Sentry breadcrumb object: https://develop.sentry.dev/sdk/event-payloads/breadcrumbs/
+ * @returns Object - A modified Sentry breadcrumb object.
+ */
+export function removeUrlsFromBreadCrumb(breadcrumb) {
+  if (breadcrumb?.data?.url) {
+    breadcrumb.data.url = hideUrlIfNotInternal(breadcrumb.data.url);
+  }
+  if (breadcrumb?.data?.to) {
+    breadcrumb.data.to = hideUrlIfNotInternal(breadcrumb.data.to);
+  }
+  if (breadcrumb?.data?.from) {
+    breadcrumb.data.from = hideUrlIfNotInternal(breadcrumb.data.from);
+  }
+  return breadcrumb;
+}
+
+/**
+ * Receives a Sentry event object and modifies it before the
+ * error is sent to Sentry. Modifications include both sanitization
+ * of data via helper methods and addition of state data from the
+ * return value of the second parameter passed to the function.
+ *
+ * @param {object} report - A Sentry event object: https://develop.sentry.dev/sdk/event-payloads/
+ * @param {Function} getState - A function that should return an object representing some amount
+ * of app state that we wish to submit with our error reports
+ * @returns Object - A modified Sentry event object.
+ */
+export function rewriteReport(report, getState) {
+  try {
+    // simplify certain complex error messages (e.g. Ethjs)
+    simplifyErrorMessages(report);
+    // remove urls from error message
+    sanitizeUrlsFromErrorMessages(report);
+    // Remove evm addresses from error message.
+    // Note that this is redundent with data scrubbing we do within our sentry dashboard,
+    // but putting the code here as well gives public visibility to how we are handling
+    // privacy with respect to sentry.
+    sanitizeAddressesFromErrorMessages(report);
+    // modify report urls
+    rewriteReportUrls(report);
+    // append app state
+    if (getState) {
+      const appState = getState();
+      if (!report.extra) {
+        report.extra = {};
       }
-    } catch (err) {
-      console.warn(err);
+      report.extra.appState = appState;
     }
-    return report;
+  } catch (err) {
+    console.warn(err);
   }
+  return report;
+}
 
-  return Sentry;
+/**
+ * Receives a Sentry event object and modifies it so that urls are removed from any of its
+ * error messages.
+ *
+ * @param {object} report - the report to modify
+ */
+function sanitizeUrlsFromErrorMessages(report) {
+  rewriteErrorMessages(report, (errorMessage) => {
+    let newErrorMessage = errorMessage;
+    const re = /(([A-z]+:\/\/)|(www\.))\S+[@:.]\S+/gu;
+    const urlsInMessage = newErrorMessage.match(re) || [];
+    urlsInMessage.forEach((url) => {
+      const urlObj = new URL(url);
```

Comment:
Hmmm. This will throw if the URL isn't valid. Luckily if this throws, it stops trying to send the whole report so we're still not leaking anything. Though it does lose us the report, which is not ideal.

Perhaps it would be safer to take the hostname from the same regex we're using to detect URLs? We have the two forward slashes as a starting point. The valid characters within a _hostname_ are quite a lot more restricted than for a URL in general, so it should be easy enough to match it.

Something like... `[-a-zA-Z0-9.]+`

Then once we have the hostname, we can look for any that match an allowlist entry or end with `.[entry]` (to capture subdomains)

Reply:
What do you think of the approach I took in 526a5d30b?

If we try to get the hostname from the regex, then we need to make the regex sophisticated enough to match the hostname but not subdomains, and also not things that look like host names in other parts of the url. There is a decent amount of safety that comes with relying on `URL`.

My approach here is just to continue to rely on that, but just catch errors and hide any url that is invalid,

Reply:
The approach taken in that commit does not match subdomains, which is a bad thing. We do want to match subdomains - otherwise we'll be filtering out URLs we use internally. `codefi.network` for example is just the root TLD we use; it's the subdomains that we're actually sending requests to.

>If we try to get the hostname from the regex, then we need to make the regex sophisticated enough to match the hostname but not subdomains, and also not things that look like host names in other parts of the url.

Hmm, wouldn't my suggestion work here? Use the regex I suggested right after the double forward slash.

Reply:
I think we have to do as done in 352d30677

Reply:
I'm not sure what you mean by that comment. But that latest commit won't work either, because now it will incorrectly match domains that have `codefi.network` as a subdomain (e.g. `codefi.network.another.domain.com`). We can't look for matches indiscriminately in the hostname, we'll have to look just at hostnames that either match exactly, or end with a period followed by the domain we're trying to match.

That's what I meant by this sentence:

>we can look for any that match an allowlist entry or end with .[entry] (to capture subdomains)

Reply:
Comparing your approach to what I suggested, it looks like this strategy will be equivalent except in the case where a matched URL is not a valid URL and causes the URL constructor to throw. In that case, we'll fallback to hiding the matched string.

This might at worst result in some of our internal URLs being hidden when they otherwise wouldn't be. I can't think of any examples offhand that this would affect. This is probably fine.